onnxruntime/core/providers/cpu/tensor/slice.cc

```diff
@@ -284,13 +330,23 @@ Status SliceImpl(OpKernelContext* ctx,
   auto* output = output_tensor.template MutableData<T>();
   const auto* output_end = output + output_tensor.Shape().Size();
 
-  SliceIterator<T> input_iterator(input_tensor, starts, output_dims, steps);
-  while (output != output_end) {
-    *output = *input_iterator;
-    ++output;
-    ++input_iterator;
+  SliceIterator<T> input_iterator =
+      flattened_output_dims != nullptr
+          ? SliceIterator<T>(input_tensor, *flattened_output_dims, starts, *flattened_output_dims, steps)
+          : SliceIterator<T>(input_tensor, starts, output_dims, steps);
```

Comment:
is it possible to specialize T with int8, int16, int32, int64 only since it's just bitwise sensitive only? so that we don't need to specialize the T for every type to reduce the binary size. like the implementation we have done in cuda 
[https://github.com/microsoft/onnxruntime/blob/5ce4d4fc494578ab3611c0e1136204d027b7d267/onnxruntime/core/providers/cuda/tensor/slice_impl.cu#L46-L68](https://github.com/microsoft/onnxruntime/blob/5ce4d4fc494578ab3611c0e1136204d027b7d267/onnxruntime/core/providers/cuda/tensor/slice_impl.cu#L46-L68)

Reply:
It's possible but I'd rather do such things in a separate PR if that's alright.